meteor/client/lib/ShowStyleVariantImportButton.tsx

```diff
@@ -0,0 +1,168 @@
+import { ShowStyleVariant } from '../../lib/collections/ShowStyleVariants'
+import { doModalDialog } from './ModalDialog'
+import { MeteorCall } from '../../lib/api/methods'
+import { logger } from '../../lib/logging'
+import React from 'react'
+import { t } from 'i18next'
+import { NoticeLevel, Notification, NotificationCenter } from './notifications/notifications'
+import { UploadButton } from './uploadButton'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faUpload } from '@fortawesome/free-solid-svg-icons'
+import { withTranslation } from 'react-i18next'
+
+interface IShowStyleVariantImportProps {
+	showStyleVariants: ShowStyleVariant[]
+}
+
+interface IShowStyleVariantImportState {
+	timestampedFileKey: number
+}
+
+export const ShowStyleVariantImportButton = withTranslation()(
+	class ShowStyleVariantImportButton extends React.Component<
+		IShowStyleVariantImportProps,
+		IShowStyleVariantImportState
+	> {
+		constructor(props: IShowStyleVariantImportProps) {
+			super(props)
+
+			this.state = {
+				timestampedFileKey: Date.now(),
+			}
+		}
+
+		private getDuplicatedShowStyleVariantAmount(showStyleVariant: ShowStyleVariant): number {
+			const importedName = showStyleVariant.name.split(' ')[0]
+			return this.props.showStyleVariants.filter((variant: ShowStyleVariant) => {
+				const existingName = variant.name.split(' ')[0]
+				return existingName === importedName
+			}).length
+		}
+
+		private importShowStyleVariants(event: React.ChangeEvent<HTMLInputElement>): void {
+			const file = event.target.files?.[0]
+			if (!file) {
+				return
+			}
+
+			const reader = new FileReader()
+
+			reader.onload = () => {
+				this.setState({
+					timestampedFileKey: Date.now(),
+				})
+				const fileContents = reader.result as string
+
+				const newShowStyleVariants: ShowStyleVariant[] = []
+				try {
+					JSON.parse(fileContents).map((showStyleVariant: ShowStyleVariant) =>
+						newShowStyleVariants.push(showStyleVariant)
+					)
+					if (!Array.isArray(newShowStyleVariants)) {
+						throw new Error('Imported file did not contain an array')
+					}
+				} catch (error) {
+					NotificationCenter.push(
+						new Notification(
+							undefined,
+							NoticeLevel.WARNING,
+							t('Failed to import new showstyle variants: {{errorMessage}}', { errorMessage: error + '' }),
+							'VariantSettings'
+						)
+					)
+					return
+				}
+
+				this.importShowStyleVariantsFromArray(newShowStyleVariants)
+			}
+			reader.readAsText(file)
+		}
+
+		private importShowStyleVariantsFromArray(importedShowStyleVariants: ShowStyleVariant[]): void {
+			importedShowStyleVariants.forEach((showStyleVariant: ShowStyleVariant, index: number) => {
+				const rank = this.props.showStyleVariants.length
+				showStyleVariant._rank = rank + index
+				if (this.showStyleVariantAlreadyExists(showStyleVariant)) {
```

Comment:
Minor thing (and somewhat comes down to preference):
Instead of writing the else block you could just return inside the if-body:
```
if (this.showStyleVariantAlreadyExists(...)) {
  this.provideImportOptions(...)
  return
}
this.importShowStyleVariant
```

Reply:
I've implemented the suggestion made by Rasmus as it looks more efficient without losing readability.